texar/torch/data/data/data_iterators.py

```diff
@@ -549,7 +555,7 @@ def __iter__(self) -> Iterator[Batch]:
         return self.get_iterator()
 
     def __len__(self):
-        return len(self._datasets[self._validate_dataset_name(None)])
+        return len(self._datasets[self._validate_dataset_name(self._current_dataset_name)])
```

Comment:
Sorry but I don't think you've solved all of the lint errors. Can you only keep the change on this line?

Also, I think this line is too long. Our line limit is 80 characters. Could you wrap this?